V1.3 show group

nbvannhi opened the pull request:
Tests will be implemented in a separate PR.

This PR will be ready for a complete review once displaying statistics graph for groups feature is implemented as some code is commented out now to be able to compile.

Please do note that the branch `v1.3-show-INDEX` is merged into this branch `v1.3-show-group` so some changes belong to the show INDEX feature :(

nbvannhi committed:
Implement show group info

nbvannhi committed:
Add stub method in AddCommandTest to resolve compile error

ganhongyao commented:
@nbvannhi Could you merge the latest master into this branch so that there are lesser diffs? Thanks!

nbvannhi committed:
Merge branch 'master' into v1.3-show-group

# Conflicts:
#	src/main/java/seedu/address/logic/commands/ShowCommand.java

ganhongyao commented on src/main/java/seedu/address/logic/commands/ShowCommand.java:
```diff
@@ -145,6 +155,24 @@ private CommandResult showAssessment(Model model) throws CommandException {
+        // uncomment when merging with graph display
+        // GroupStatistics statistics = new GroupStatistics(matchedGroup);
+        // return new CommandResult(MESSAGE_SUCCESS, info, statistics.toLineChart());
+        return new CommandResult(MESSAGE_SUCCESS, info); // delete when merging
```
Is this referring to this PR https://github.com/AY2122S1-CS2103T-W08-2/tp/pull/178 ?

nbvannhi replied:
Yep!

ganhongyao commented on src/main/java/seedu/address/logic/commands/ShowCommand.java:
```diff
@@ -210,17 +243,17 @@ public void setAssessment(Assessment assessment) {
      * Stores info of a student or an assessment.
      */
     public static class Info {
-        private Index index;
         private Student student;
         private Assessment assessment;
-
-        public Info(Index index) {
-            setIndex(index);
-        }
```
I think you accidentally removed the changes in the other PR while merging

nbvannhi replied:
Ah actually no. When I did this show graph part I realised that the `Index` field is redundant since `Info` is responsible for showing stats either a `Student`, an `Assessment` or a `Group`, but not an `Index`. And I forgot to switch branch and make a separate commit for that..

ganhongyao replied:
Oh right okay!

ganhongyao commented on src/main/java/seedu/address/logic/parser/ShowCommandParser.java:
```diff
@@ -57,7 +58,9 @@ public ShowCommand parseByPrefixes(ArgumentMultimap argMultimap) throws ParseExc
-                : new ShowCommand(ParserUtil.parseAssessment(argMultimap.getValue(PREFIX_ASSESSMENT).get()));
+                : argMultimap.getValue(PREFIX_ASSESSMENT).isPresent()
+                ? new ShowCommand(ParserUtil.parseAssessment(argMultimap.getValue(PREFIX_ASSESSMENT).get()))
+                : new ShowCommand(ParserUtil.parseGroup(argMultimap.getValue(PREFIX_GROUP).get()));
```
I don't think we can assume that when all other prefixes are not present group will definitely be present. Currently, if I just execute "show", I get an exception because we call `get` on `argMultimap.getValue(PREFIX_GROUP)`, which is not present.

nbvannhi replied:
Resolved!

ganhongyao commented on src/main/java/seedu/address/logic/commands/ShowCommand.java:
```diff
@@ -33,48 +35,56 @@
     public ShowCommand(Index index) {
-        this(index, null, null, null);
+        this(index, null, null, null, null);
     }
```
Would it be cleaner if we just do `setIndex(index)` in this constructor instead of calling the overloaded constructor with `null`s? Same for the others

nbvannhi replied:
Make sense! Resolving this

nbvannhi replied:
Resolved!

nbvannhi committed:
Remove sharing between constructors

nbvannhi committed:
Fix exception not thrown bug when invalid `show` command executed

When `show` command is executed with empty preamble,
a `ParseException` should be thrown.

zhing22 commented on src/main/java/seedu/address/logic/commands/ShowCommand.java:
```diff
@@ -145,6 +159,24 @@ private CommandResult showAssessment(Model model) throws CommandException {
+
+        Info info = new Info(matchedGroup);
+        // uncomment when merging with graph display
+        // GroupStatistics statistics = new GroupStatistics(matchedGroup);
```
pass in model as parameters
```suggestion
        GroupStatistics statistics = new GroupStatistics(matchedGroup, model);
```

jonas-chow committed:
Add functionality to save graph into file

nbvannhi committed:
Merge branch 'v1.3b-grouplinechart' of https://github.com/AY2122S1-CS2103T-W08-2/tp into v1.3-show-group

nbvannhi committed:
Complete implementing show group

nbvannhi committed:
Fix checkstyle

nbvannhi committed:
Merge branch 'master' into v1.3-show-group

nbvannhi committed:
Remove (again) unused methods due to merging

nbvannhi commented:
@ganhongyao it's ready for review now!

nbvannhi committed:
Reset filtered student list when showing assessment info

nbvannhi committed:
Merge branch 'v1.3-export-graph' of https://github.com/AY2122S1-CS2103T-W08-2/tp into test

# Conflicts:
#	src/main/java/seedu/address/logic/commands/ShowCommand.java
#	src/main/java/seedu/address/logic/parser/ShowCommandParser.java

nbvannhi committed:
Merge branch 'test' into v1.3-show-group

nbvannhi committed:
Fix checkstyle (after adapting changes from v1.3-export-graph)

ganhongyao commented:
LGTM! Nice
